### PR TITLE
Checkbox style changes

### PR DESCRIPTION
**Pull request by yuderekyu**
### Summary
Addresses #1126, #1130, #1136
* Removes select-container css class.
* Increase padding for labeltext when checkbox is resized for mobile devices.
* Changes focus style from `box-shadow` to `border-width`/`border-color`
* Adds `transition` to the unchecked state. This allows animations for hover and focus when the checkbox is in an unchecked state.
* Increase specificity of mobile, disabled, hidden, and inline css classes.
* Spread custom props  

### Deployment Link
https://terra-core-deployed-pr-1137.herokuapp.com/static/#/site/form-checkbox

### CODE REVIEW

File: packages/terra-form-checkbox/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,15 @@ ChangeLog
 
 (Unreleased)
 ----------
+### Added
+* Increase padding on mobile resize
```
Comment (StephenEsser) [inline packages/terra-form-checkbox/CHANGELOG.md]: Increasing the padding might belong in the `Changed` section

File: packages/terra-form-checkbox/src/Checkbox.scss [hunk 2]
```diff
@@ -120,19 +115,12 @@
     .label:hover {
       .label-text::before {
         border-color: var(--terra-form-checkbox-hover-border-color, none);
-
-        [dir=rtl] & {
```
Comment (bjankord) [inline packages/terra-form-checkbox/src/Checkbox.scss]: What's the logic around removing the dir selectors? I thought these were needed to fix some issues where initial value is non-bidirectional, e.g. `none` and then the themed value includes styles which would be bidirectional?

**Comment by yuderekyu** [comment]
After offline discussion with @neilpfeiffer: margin-top now defaults to 0 and margin-bottom is added as a themeable variable.

**Comment by mmalaker** [comment]
Functional verification completed. The checkbox examples display and function correctly with the new style changes.

**Pull request merged**